crates/solidity/outputs/cargo/crate/src/backend/binder/definitions.rs

```diff
@@ -139,121 +133,142 @@ pub enum StateVariableVisibility {
+            Self::YulParameter(parameter_definition) => parameter_definition.ir_node.id(),
+            Self::YulVariable(variable_definition) => variable_definition.ir_node.id(),
         }
     }
 
     pub fn identifier(&self) -> &Rc<TerminalNode> {
         match self {
-            Self::Constant(constant_definition) => &constant_definition.identifier,
-            Self::Contract(contract_definition) => &contract_definition.identifier,
-            Self::Enum(enum_definition) => &enum_definition.identifier,
-            Self::EnumMember(enum_member_definition) => &enum_member_definition.identifier,
-            Self::Error(error_definition) => &error_definition.identifier,
-            Self::Event(event_definition) => &event_definition.identifier,
-            Self::Function(function_definition) => &function_definition.identifier,
-            Self::Import(import_definition) => &import_definition.identifier,
-            Self::ImportedSymbol(symbol_definition) => &symbol_definition.identifier,
-            Self::Interface(interface_definition) => &interface_definition.identifier,
-            Self::Library(library_definition) => &library_definition.identifier,
-            Self::Modifier(modifier_definition) => &modifier_definition.identifier,
-            Self::Parameter(parameter_definition) => &parameter_definition.identifier,
-            Self::StateVariable(state_variable_definition) => &state_variable_definition.identifier,
-            Self::Struct(struct_definition) => &struct_definition.identifier,
-            Self::StructMember(struct_member_definition) => &struct_member_definition.identifier,
-            Self::TypeParameter(parameter_definition) => &parameter_definition.identifier,
-            Self::UserDefinedValueType(udvt_definition) => &udvt_definition.identifier,
-            Self::Variable(variable_definition) => &variable_definition.identifier,
-            Self::YulFunction(function_definition) => &function_definition.identifier,
-            Self::YulLabel(label_definition) => &label_definition.identifier,
-            Self::YulParameter(parameter_definition) => &parameter_definition.identifier,
-            Self::YulVariable(variable_definition) => &variable_definition.identifier,
+            Self::Constant(constant_definition) => &constant_definition.ir_node.name,
+            Self::Contract(contract_definition) => &contract_definition.ir_node.name,
+            Self::Enum(enum_definition) => &enum_definition.ir_node.name,
+            Self::EnumMember(enum_member_definition) => &enum_member_definition.ir_node,
+            Self::Error(error_definition) => &error_definition.ir_node.name,
+            Self::Event(event_definition) => &event_definition.ir_node.name,
+            Self::Function(function_definition) => {
+                // Function definitions are only created for *named* functions
+                function_definition.ir_node.name.as_ref().unwrap()
```

Comment:
Not for this PR, but performance-wise we may want to consider [going `unsafe`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_unchecked) in these cases (also worth checking how big a difference it makes).

```suggestion
                unsafe { function_definition.ir_node.name.as_ref().unwrap_unchecked() }
```